sp-files/.github/workflows/markdown-style-checks.yml

```diff
@@ -0,0 +1,15 @@
+name: "Linter for Markdown"
+
+on: [push, pull_request]
+
+jobs:
+  markdown-lint:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0     
+      - uses: DavidAnson/markdownlint-cli2-action@v16
+        with:
+          config: ".markdownlint.json"
```

Comment:
I assume you use this file in the root of the repo, while your config lays in the subfolder

Reply:
The checks in the root folder of the starter pack are for verifying the functionality of the starter pack. Starter pack based documentation should have these files in their root folders, and this is done by the [init.sh](https://github.com/canonical/sphinx-docs-starter-pack/blob/0fadd8f8f48a38270cc00c8f30861c2efdbc0d5a/init.sh#L136) script.

Reply:
1. Change dir for `.markdownlint.json` to `sp-docs/.sphinx`
2. Add line to [this section of the init script](https://github.com/canonical/sphinx-docs-starter-pack/blob/0fadd8f8f48a38270cc00c8f30861c2efdbc0d5a/init.sh#L56):
```
[ "$install_directory" != "." ] && sed -i "s/.markdownlint.json/$install_directory\/.sphinx\/markdown-style-checks.yml/g" "$temp_directory/sp-files/.github/workflows/markdown-style-checks.yml"
"
```

Reply:
I am assuming you mean:
[ "$install_directory" != "." ] && sed -i "s/.sphinx\/.markdownlint.json/$install_directory\/.sphinx\/**.markdownlint.json/** "$temp_directory/sp-files/.github/workflows/markdown-style-checks.yml"

and not 
[ "$install_directory" != "." ] && sed -i "s/.sphinx\/.markdownlint.json/$install_directory\/.sphinx\/**markdown-style-checks.yml/g**" "$temp_directory/sp-files/.github/workflows/markdown-style-checks.yml"

Reply:
Also, `sp-docs/.sphinx` is a symbolic link to `sp-files/.sphinx`, so the `.markdownlint.json` file gets added to `sp-files/.sphinx` (and is also in turn available in `sp-docs/.sphinx`).

Reply:
Right on both counts! Thanks for the correction :)